### ISSUE TITLE
MonHun Nikki - Screen dark overlay (Android)

### ISSUE THREAD

**Issue by Saramagrean** [hrydgard/ppsspp#10726]
• What happens?
MonHun Nikki Pokapoka Airu Mura (ULJM-05710) Screen dark overlay since v1.0.1-432-g42bcea7 (20150417), same issue on latest build (v1.5.4-677) with OpenGL only, Vulkan is work fine.

v1.0.1-432-g42bcea7 use Default settings.
![screenshot_20180316-215030](https://user-images.githubusercontent.com/26355472/37527900-bb2159d8-2965-11e8-9e49-c84f34a20be1.png)

• What *should* happen?
Last work v1.0.1-431-ga9ff7c2 (20150417) & latest git build with Vulkan.

v1.0.1-431-ga9ff7c2 use Default setting
![screenshot_20180316-214941](https://user-images.githubusercontent.com/26355472/37528005-0498e478-2966-11e8-9192-a7af9b9adac2.png)

• What hardware and operating system are you running PPSSPP on? GPU might matter if it's a graphical issue.

Huawei P9+ (Kirin 955, Mali T880)

I tried on PC is work flawless all backend. 

Sorry for my terrible english. 😂

**Comment by unknownbrackets** [comment]
If 42bcea7 is the first bad build, and a9ff7c2 is the last working one, then it must be the difference between `/ 65535.0f` and `* (1.0f / 65535.0f)`?  I'm guessing that means it's something that is near outside the depth range.

Vulkan probably works because of accurate depth, then.

-[Unknown]

**Comment by unknownbrackets** [comment]
Does this still happen?  If so, could you try exporting a GE frame dump?  These help a lot.

See here for instructions - it's not hard and works on Android too:
https://github.com/hrydgard/ppsspp/wiki/How-to-create-a-frame-dump

You can zip that and then drag and drop it into a reply here.

-[Unknown]

**Comment by Emulatorer** [comment]
@Saramagrean try enable the 4xMSAA in your phone developers options and see if the dark overlay is fixed?

**Comment by Saramagrean** [comment]
Nope. 
![Screenshot_PPSSPP_20190814-180604](https://user-images.githubusercontent.com/26355472/63017361-56c10780-bec0-11e9-86e0-bcb4dafbff65.png)

**Comment by hrydgard** [comment]
We should probably revert those to actual divisions (or do the multiplications in double precision).

**Comment by Emulatorer** [comment]
Yeah same happen to me I thought 4XMSAA can help the dark overlay issue 😅
![Screenshot_2019-08-14-20-23-43](https://user-images.githubusercontent.com/37603562/63020993-1ec2c000-bed2-11e9-925d-66b659044327.png)

**Comment by Saramagrean** [comment]
Update GE dump. (v1.9.3-973-g2ba65c831)
https://drive.google.com/file/d/11PP9E63pkRRYly9TZvC1Xj0jvxzvFwOF/view?usp=drivesdk

**Comment by Saramagrean** [comment]
Try again with v1.10.3 latest git build (461), It seem like fixed. :)
![Screenshot_2020-08-19-05-37-51-928_org ppsspp ppsspp](https://user-images.githubusercontent.com/26355472/90573897-c0fe3d80-e1e1-11ea-87bd-46830c10816e.jpg)

**Comment by Gamemulatorer** [comment]
Looks like not on my phone still dark
![Screenshot_2020-08-19-07-35-13](https://user-images.githubusercontent.com/37603562/90575728-a2527380-e1ee-11ea-827a-1b63dcf23b1d.png)

**Comment by Saramagrean** [comment]
Oh, maybe this issue happen in Mali only? my Huawei fall in water can't turn on 😂, so I use other device (Redmi Note 8) to test it.

**Comment by Panderner** [comment]
![Screenshot_2020-08-19-09-34-37-43_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/90582084-7c819a80-e1ff-11ea-9f5c-e0f66108553b.png)
![Screenshot_2020-08-19-09-35-23-25_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/90582093-80adb800-e1ff-11ea-8f26-1bc0b2c1cf7e.png)
The game works fine for me running on Realme C2

**Comment by Gamemulatorer** [comment]
Probably my phone is gles2 and it does not support depth/z 😔

**Comment by Panderner** [comment]
and same goes to my POCO M3 in OpenGL.
![Screenshot_2021-02-05-22-31-51-399_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/107047114-69d93d00-6802-11eb-981b-5594894ea506.jpg)

![Screenshot_2021-02-05-22-34-12-078_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/107047073-5f1ea800-6802-11eb-93da-d21bca4df9aa.jpg)

**Comment by Gamemulatorer** [comment]
Maybe this issue only happen on mali gpu's?
### OpenGL
![Screenshot_2021-05-03-18-04-32-781_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/116864504-871f7b00-ac3a-11eb-8f86-1fc0085402ed.jpg)
### Vulkan
![Screenshot_2021-05-03-18-04-20-295_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/116864537-97cff100-ac3a-11eb-87ec-c633e2ef247a.jpg)

**Comment by Gamemulatorer** [comment]
> ![Screenshot_2020-08-19-09-34-37-43_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/90582084-7c819a80-e1ff-11ea-9f5c-e0f66108553b.png)
> ![Screenshot_2020-08-19-09-35-23-25_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/90582093-80adb800-e1ff-11ea-8f26-1bc0b2c1cf7e.png)
> The game works fine for me running on Realme C2 in OpenGL

Realme C2 GPU is PowerVR?

**Comment by Panderner** [comment]
> > ![Screenshot_2020-08-19-09-34-37-43_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/90582084-7c819a80-e1ff-11ea-9f5c-e0f66108553b.png)
> > ![Screenshot_2020-08-19-09-35-23-25_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/90582093-80adb800-e1ff-11ea-8f26-1bc0b2c1cf7e.png)
> > The game works fine for me running on Realme C2 in OpenGL
> 
> Realme C2 GPU is PowerVR?

Umm yes Realme C2 is a PowerVR GE8320 device but it doesn't support vulkan.

**Comment by Gamemulatorer** [comment]
> > > ![Screenshot_2020-08-19-09-34-37-43_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/90582084-7c819a80-e1ff-11ea-9f5c-e0f66108553b.png)
> > > ![Screenshot_2020-08-19-09-35-23-25_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/90582093-80adb800-e1ff-11ea-8f26-1bc0b2c1cf7e.png)
> > > The game works fine for me running on Realme C2 in OpenGL
> > 
> > 
> > Realme C2 GPU is PowerVR?
> 
> Umm yes Realme C2 is a PowerVR GE8320 device but it doesn't support vulkan.

So technically this is a Mali GPU's issue only

**Comment by unknownbrackets** [comment]
This draws initially with 808080FF as the color, which results in a dimmed image.  This draw happens with Z=65000, and max=65000.

Then it blends a dst+src blend with color FFFFFFFF, so in other words doubling the dst color values.  This draw happens with z=100, and minz=100 as well.

Viewport Z is 32550 -/+ 32450.

So my guess is this is related to accurate depth only,, and some sort of rounding issue with that indeed.

-[Unknown]

**Comment by buymyhubs** [comment]
Adding this title to [DepthRangeHack] in compat.ini seems to resolve this issue on Mali GPUs using OpenGL. See PR #21000

**Comment by buymyhubs** [comment]
This issue can probably be closed as it's been addressed in #21000. I guess the issue didn't link
